### ISSUE TITLE
LR scheduler resume applies the old weight-decay schedule and treats num_steps as a delta

### ISSUE THREAD

**Issue by kabirvashisht4-glitch** [NVIDIA-NeMo/Automodel#3882]
## Describe the bug

`OptimizerParamScheduler.load_state_dict` writes the resolved learning rate and
weight decay into `optimizer.param_groups` **before** it has restored the
weight-decay fields. On resume the learning rate comes from the checkpoint's
schedule while the weight decay comes from the constructor's, and nothing
recomputes it until the next `scheduler.step()`.

The training loop applies the optimizer update first
(`nemo_automodel/recipes/llm/train_ft.py`: `opt.step()` at L1275,
`scheduler.step(1)` at L1284), so the stale weight decay is used for a real
parameter update.

The ordering in `nemo_automodel/components/optim/scheduler.py`:

```python
self.lr_decay_style = self._check_and_set(...)   # LR fields restored

num_steps = state_dict["num_steps"]
self.step(increment=num_steps)                   # <- resolves + WRITES lr and wd

if "start_wd" in state_dict:                     # <- wd fields restored AFTER
    self.start_wd = self._check_and_set(...)
    self.end_wd = self._check_and_set(...)
    self.wd_incr_steps = self._check_and_set(...)
    self.wd_incr_style = self._check_and_set(...)
```

Every learning-rate field is restored before `step()`; every weight-decay field
is restored after it. The asymmetry looks unintentional.

This matters precisely when the checkpoint and the constructor disagree, which
is the case `use_checkpoint_opt_param_scheduler=True` (the default) exists to
serve — `_check_and_set` is written to let checkpoint values win.

## Second problem in the same call: the step count is folded in, not set

```python
self.step(increment=num_steps)
```

`step()` does `self.num_steps += increment`, but `num_steps` in the state dict
is an absolute position. That is only correct when the scheduler has counted
nothing yet. Loading twice, or loading a scheduler that has already advanced,
lands past the checkpoint's step and silently shifts the rest of the schedule.

## Steps/Code to reproduce bug

```python
import torch
from nemo_automodel.components.optim.scheduler import OptimizerParamScheduler

def build(**overrides):
    p = [torch.nn.Parameter(torch.zeros(1))]
    opt = torch.optim.SGD(p, lr=1.0, weight_decay=0.0)
    kw = dict(optimizer=opt, init_lr=0.0, max_lr=1.0, min_lr=0.0,
              lr_warmup_steps=10, lr_decay_steps=100, lr_decay_style="cosine",
              start_wd=0.0, end_wd=0.1, wd_incr_steps=100, wd_incr_style="linear")
    kw.update(overrides)
    return OptimizerParamScheduler(**kw), opt

ref, ropt = build()
ref.step(50)
sd = ref.state_dict()
print("reference @50 :", "lr=", round(ropt.param_groups[0]["lr"], 6),
      "wd=", round(ropt.param_groups[0]["weight_decay"], 6))

# resume with a different weight-decay schedule in the constructor
res, ropt2 = build(end_wd=0.9, wd_incr_steps=1000)
res.load_state_dict(sd)
print("after load    :", "lr=", round(ropt2.param_groups[0]["lr"], 6),
      "wd=", round(ropt2.param_groups[0]["weight_decay"], 6))
print("restored fields: end_wd=", res.end_wd, "wd_incr_steps=", res.wd_incr_steps)
print("wd the restored schedule implies:", round(res.get_wd(), 6))

a, _ = build(); a.load_state_dict(sd); print("load once  num_steps=", a.num_steps)
b, _ = build(); b.load_state_dict(sd); b.load_state_dict(sd); print("load twice num_steps=", b.num_steps)
```

```
reference @50 : lr= 0.586824 wd= 0.05
after load    : lr= 0.586824 wd= 0.045      <-- constructor's schedule, not the checkpoint's
restored fields: end_wd= 0.1 wd_incr_steps= 100
wd the restored schedule implies: 0.05

load once  num_steps= 50
load twice num_steps= 100                   <-- absolute count folded in as a delta
```

The learning rate resumes correctly at `0.586824`. The weight decay is
`0.045` — that is `50/1000 * 0.9` from the constructor — where the restored
schedule implies `50/100 * 0.1 = 0.05`. The scheduler's own `get_wd()` already
returns `0.05`; only the value handed to the optimizer is wrong.

## Expected behavior

`load_state_dict` should restore every field before applying any of them, so the
learning rate and the weight decay written into `optimizer.param_groups` both
come from the same (checkpoint) schedule. The step count should be treated as
the absolute position it is, making the load idempotent.

## Environment overview

- `main` @ `8a646a73`
- CPU only, no GPU needed to reproduce.

## Additional context

Existing coverage in `tests/unit_tests/optim/test_scheduler.py` asserts the
restored *attributes* (`scheduler.end_wd == 0.2`, etc.) but never asserts what
`load_state_dict` wrote into `optimizer.param_groups`, which is where the
divergence shows up. `test_load_state_dict_use_checkpoint_true` also documents
the increment behaviour in a comment — `# num_steps from checkpoint + current 0`
— which holds only because that scheduler is freshly constructed.
